src/internal/connector/exchange/exchange_data_collection.go

```diff
@@ -139,33 +144,83 @@ func (col *Collection) populateByOptionIdentifier(
 		return
 	}
 
+	limitCh := make(chan struct{}, urlPrefetchChannelBufferSize)
+	defer close(limitCh)
+
+	type uerr struct {
+		user string
+		err  error
+	}
+
+	errCh := make(chan uerr, len(col.jobs))
+	defer close(errCh)
+
+	ffail := int64(0)
 	for _, identifier := range col.jobs {
-		response, err := query(ctx, col.service, user, identifier)
-		if err != nil {
-			errs = support.WrapAndAppendf(user, err, errs)
+		if ffail > 0 {
+			break
+		}
+		limitCh <- struct{}{}
+
+		wg.Add(1)
+
+		go func(identifier string) {
+			defer func() { <-limitCh }()
+			defer wg.Done()
+
+			var (
+				response absser.Parsable
+				err      error
+			)
```

Comment:
There are several different code declarations. Could it be simplified into the top of the function?

Reply:
These are the only declarations in the func called by goroutines.